test/src/crypto.js

```diff
@@ -153,4 +153,24 @@ describe('crypto', () => {
       expect(await crypto.decrypt(encrypted)).to.be('I am a string');
     });
   });
+
+  describe('Using AAD', () => {
+    it('should decryptable with same AAD', async () => {
+      const encrypted = await crypto.encrypt('Hello World!', 'some aad');
+
+      const decrypted = await crypto.decrypt(encrypted, 'some aad');
+      expect(decrypted).to.be.equal('Hello World!');
+    });
+
+    it('should not be decryptable without aad used in encryption', async () => {
+      const encrypted = await crypto.encrypt('Hello World!', 'some aad');
+      try {
+        const decrypted = await crypto.decrypt(encrypted);
+        throw new Error(`Decryption should fail but value was ${decrypted}`);
+      }catch(err) {
+        expect(err).to.be.an(Error);
+        expect(err.message).to.match(/Unsupported state or unable to authenticate data/);
+      }
+    });
```

Comment:
Sorry for the drive-by review, but I would feel more comfortable if there was an additional test with the AAD being a single byte different (not just missing). 
```
      const encrypted = await crypto.encrypt('some text', '123456789');
      try {
        const decrypted = await crypto.decrypt(encrypted, '123456780' );
```

Reply:
No apology needed! I considered making the same request, but didn’t initially because it seemed like it’d be testing node’s crypto library itself. However, it would also make me feel a more comfortable.

Reply:
Crypto APIs are notoriously for being confusing to use and behaving in unexpected ways. So, something like this is at risk of looking like the AAD value would be verified in the `setAAD` method, but actually it turns out that you need to call `verifyAAD` afterwards, or something like that.
So, I see this as testing that we're calling node's crypto correctly, rather than testing that the library works. If we wanted to do the latter, we'd need a bunch more tests, particularly with larger data sizes.

Reply:
@tvernum No need to apologize the more eyes the better. I added a test with the one byte difference. It passes so that is comforting. I don't see any methods or process in node-crypto to do verification of AAD.